### PR TITLE
Improve Microsoft login flow

### PR DESCRIPTION
**Pull request by Trial97**
Stolen from TheKodeToad from here: https://discord.com/channels/1031648380885147709/1031823065937629267/1171229791513493544

Please don't hesitate to suggest any text changes.
I kept the old behavior in case the user browser was not open or something was unclear.

Also tested and it works.

### CODE REVIEW

File: launcher/ui/dialogs/MSALoginDialog.cpp [hunk 1]
```diff
@@ -105,8 +105,17 @@ void MSALoginDialog::showVerificationUriAndCode(const QUrl& uri, const QString&
...
     QString linkString = QString("<a href=\"%1\">%2</a>").arg(urlString, urlString);
-    ui->label->setText(
-        tr("<p>Please open up %1 in a browser and put in the code <b>%2</b> to proceed with login.</p>").arg(linkString, code));
+    if (urlString == "https://www.microsoft.com/link" && !code.isEmpty()) {
+        urlString += QString("?otc=%1").arg(code);
+        DesktopServices::openUrl(urlString);
+        ui->label->setText(
+            tr("<p>Please login in the opened browser. If no browser was opened please follow the following instructions:</p>"
+               "<p>Please open up %1 in a browser and put in the code <b>%2</b> to proceed with login.</p>")
+                .arg(linkString, code));
```
Comment (TayouVR) [inline launcher/ui/dialogs/MSALoginDialog.cpp]:
```suggestion
        ui->label->setText(
            tr("<p>Please login in the opened browser. If no browser was opened please open up %1 in "
               "a browser and put in the code <b>%2</b> to proceed with login.</p>")
                .arg(linkString, code));
```

**Review by TayouVR — APPROVED** [review]
text could be a bit nicer, maybe like my suggestion below, but its fine either way (I want this entire dialog to get an overhaul some day x3)

File: launcher/ui/dialogs/MSALoginDialog.cpp [hunk 2]
```diff
@@ -105,8 +105,16 @@ void MSALoginDialog::showVerificationUriAndCode(const QUrl& uri, const QString&
 
     QString urlString = uri.toString();
     QString linkString = QString("<a href=\"%1\">%2</a>").arg(urlString, urlString);
-    ui->label->setText(
-        tr("<p>Please open up %1 in a browser and put in the code <b>%2</b> to proceed with login.</p>").arg(linkString, code));
+    if (urlString == "https://www.microsoft.com/link" && !code.isEmpty()) {
+        urlString += QString("?otc=%1").arg(code);
+        DesktopServices::openUrl(urlString);
+        ui->label->setText(tr("<p>Please login in the opened browser. If no browser was opened please open up %1 in "
```
Comment (getchoo) [inline launcher/ui/dialogs/MSALoginDialog.cpp]:
```suggestion
        ui->label->setText(tr("<p>Please login in the opened browser. If no browser was opened, please open up %1 in "
```

**Review by getchoo — CHANGES_REQUESTED** [review]
otherwise lgtm :)

**Review by getchoo — APPROVED** [review]

**Comment by getchoo** [comment]
~~it would be so cool if we had automerge~~

**Review by Scrumplex — APPROVED** [review]

**Comment by TheKodeToad** [comment]
wouldn't it be better to have a button instead of opening the browser straight away

**Comment by Trial97** [comment]
Not really as I want to simplify the process. not keep it the same.
Also a little late to the party as it was merged :(.
Feel free to open a PR where you can improve that behavior.

**Comment by TheKodeToad** [comment]
ig it's fine

**Pull request merged**